src/Nethermind/Nethermind.Consensus.AuRa/AuRaBlockProcessor.cs

```diff
@@ -48,18 +51,38 @@ public class AuRaBlockProcessor : BlockProcessor
             ITxPool txPool,
             IReceiptStorage receiptStorage,
             ILogManager logManager,
-            IAuRaBlockProcessorExtension auRaBlockProcessorExtension)
+            IAuRaBlockProcessorExtension auRaBlockProcessorExtension,
+            ITxPermissionFilter txFilter = null)
             : base(specProvider, blockValidator, rewardCalculator, transactionProcessor, stateDb, codeDb, stateProvider, storageProvider, txPool, receiptStorage, logManager)
         {
             _auRaBlockProcessorExtension = auRaBlockProcessorExtension ?? throw new ArgumentNullException(nameof(auRaBlockProcessorExtension));
+            _logger = logManager?.GetClassLogger<AuRaBlockProcessor>() ?? throw new ArgumentNullException(nameof(logManager));
+            _txFilter = txFilter;
         }
 
         protected override TxReceipt[] ProcessBlock(Block block, IBlockTracer blockTracer, ProcessingOptions options)
         {
+            ValidateTxs(block);
             _auRaBlockProcessorExtension.PreProcess(block, options);
             var receipts = base.ProcessBlock(block, blockTracer, options);
             _auRaBlockProcessorExtension.PostProcess(block, receipts, options);
             return receipts;
         }
+
+        private void ValidateTxs(Block block)
+        {
+            if (_txFilter != null)
+            {
+                for (int i = 0; i < block.Transactions.Length; i++)
+                {
+                    var tx = block.Transactions[i];
+                    if (!_txFilter.IsAllowed(tx, block.Header))
+                    {
+                        if (_logger.IsError) _logger.Error($"Proposed block is not valid {block.ToString(Block.Format.FullHashAndNumber)}. {tx.ToShortString()} doesn't have required permissions.");
+                        throw new InvalidBlockException(block.Hash);
```

Comment:
what if such a block is sent to you - will it cause the blockchain processor to fail forever? (or same as wrong state root just switch the branch)

Reply:
Good point. So I need to reject this block. But because I need state from previous block I cannot do it anywhere earlier. Do you have any suggestions?

Reply:
It should throw as its incorrect block